compiler/src/main/scala/org/scalajs/core/compiler/Compat210Component.scala

```diff
@@ -58,6 +66,25 @@ trait Compat210Component {
     }
   }
 
+  // Impl classes disappeared in 2.12.0-M4
+
+  lazy val scalaUsesImplClasses: Boolean =
+    definitions.SeqClass.implClass != NoSymbol // a trait we know has an impl class
+
+  implicit final class StdTermNamesCompat(self: global.nme.type) {
+    val IMPL_CLASS_SUFFIX = "$class"
+
+    def isImplClassName(name: Name): Boolean =
+      name.endsWith(IMPL_CLASS_SUFFIX)
```

Comment:
It certainly should, in practice.

What I did in general in this PR was to follow the overall "contract" of `Compat210` to maintain the same API with the same behavior over a large set of versions. Which led me to "re-implement" some functionality that was lost in upstream.

In this case, it's probably also valid to reimplement differently those methods, knowing no impl class ever exists when those come in. It's a slightly different way of looking at `Compat210`, which is probably equally valid.

I guess your preference goes to the second "contract"? I don't have any strong opinion on this.